### PR TITLE
Solr query sorting param

### PR DESCRIPTION
**Pull request by jesperlauridsen**
added paging.
added some solr params for better searches.
Minor bugfixes

### CODE REVIEW

File: src/components/searchresult/common/PDFSearchResult.vue [hunk 1]
```diff
@@ -60,6 +60,19 @@
...
-    }
+    },
+    methods: {
+      checkForSnippetAvailability(query) {
+        if(query === '*.*' || query === '*:*' || query.includes('&pt=')) {
+          return false;
+        }
+        else {
+          return true;
+        }
```
Comment (jorntx) [inline src/components/searchresult/common/PDFSearchResult.vue:71]: I actually stumbled upon a case where a ternary would have cleaner ;)))

File: src/components/SearchResults.vue [hunk 2]
```diff
@@ -13,14 +13,19 @@
...
+          </div>
+          <div v-if="loading === false && this.results.length > 0" class="pagingButtonContainer">
+            <button class="pagingButton back" :disabled="this.solrOptions.currentOffset <= 0" v-on:click="previousResults">previous 10</button>
+            <button class="pagingButton forward" :disabled="this.solrOptions.currentOffset + this.solrOptions.shownResultsNumber >= this.resultHits" v-on:click="nextResults">next 10</button>
+          </div>
+          <component :is="searchResultComponentName"  v-for="(item, index) in this.results" :result="item" :indexNumber="index" :queryString="queryDisplay" :key="index"  ></component>
+          <div v-if="loading === false && this.results.length > 0" class="pagingButtonContainer pageEnd">
+            <button class="pagingButton back" :disabled="this.solrOptions.currentOffset <= 0" v-on:click="previousResults">previous 10</button>
+            <button class="pagingButton forward" :disabled="this.solrOptions.currentOffset + this.solrOptions.shownResultsNumber >= this.resultHits" v-on:click="nextResults">next 10</button>
```
Comment (jorntx) [inline src/components/SearchResults.vue:27]: With this simple pager it's ok to keep it here - if we do anything more fancy with clickable page numbers etc. we will move it to seperate component.

File: src/components/AppliedFilters.vue [hunk 3]
```diff
@@ -135,6 +122,7 @@ export default {
           time = true;
           name = name.replace(/[*[\]]/g,'').replace(/TO|FROM/g, '');
```
[the remark this answers is not in the document]
Comment (jesperlauridsen) [reply src/components/AppliedFilters.vue:123]: Comment added.

File: src/components/AppliedFilters.vue [hunk 4]
```diff
@@ -135,6 +122,7 @@ export default {
...
         }
+        name.includes("&sort") ? name = name.substring(0, name.indexOf('&sort')) : null
       }
       return time === true ? this.$_deliverTimePeriodStamp(name) : name;
```
[the remark this answers is not in the document]
Comment (jesperlauridsen) [reply src/components/AppliedFilters.vue:127]: Roger. should this be the case in all places? A little longer up in this section (line 114) we have:

      if (filter.includes("&pt=") === true) {

Should this kind of code be changed aswell, in your oppinion?

File: src/components/SearchBox.vue [hunk 5]
```diff
@@ -198,11 +212,11 @@
       //Check and apply (if needed), time filters to the querystring.
       addTimeFiltersToSearchFilter(filterString) {
         if(this.time === true && filterString.includes("ff_primaryobject_year_from_i") === false && (this.timeFrom !== this.searchOptions.timeOptions.min || this.timeTo !== this.searchOptions.timeOptions.max)) {
```
[the remark this answers is not in the document]
Comment (jesperlauridsen) [reply src/components/SearchBox.vue:214]: Changed.

File: src/components/SearchBox.vue [hunk 6]
```diff
@@ -198,11 +212,11 @@
...
-          filterString += "&fq=ff_primaryobject_year_from_i:[* TO " + this.timeFrom + "]";
-          }
-          if(this.time === true && filterString.includes("ff_primaryobject_year_to_i") === false && (this.timeFrom !== this.searchOptions.timeOptions.min || this.timeTo !== this.searchOptions.timeOptions.max)) {
-            filterString += "&fq=ff_primaryobject_year_to_i:[" + this.timeTo + " TO *]";
-          }
+          filterString += "&fq=ff_primaryobject_year_from_i:[* TO " + this.timeTo + "]";
+        }
+        if(this.time === true && filterString.includes("ff_primaryobject_year_to_i") === false && (this.timeFrom !== this.searchOptions.timeOptions.min || this.timeTo !== this.searchOptions.timeOptions.max)) {
```
[the remark this answers is not in the document]
Comment (jesperlauridsen) [reply src/components/SearchBox.vue:217]: Changed.

File: src/components/SearchResults.vue [hunk 7]
```diff
@@ -13,14 +13,19 @@
...
-             <span class="numbersFound">{{ resultHits || "0" }}</span> matches was found in
-               <span class="numbersFound"> {{ resultMatches || "0" }}</span>
-               <span v-if="resultHits > 1 || resultHits === null || resultHits === 0"> pdfs</span>
-               <span v-if="resultHits === 1"> pdf</span>
-           </div>
-           
-
-              <component :is="searchResultComponentName"  v-for="(item, index) in this.results" :result="item" :indexNumber="index" :queryString="queryDisplay" :key="index"  ></component>
+             <span class="numbersFound">{{ resultHits || "0" }}</span> matches was found.
+             <span v-if="loading === false && this.results.length > 0">Showing hit
```
[the remark this answers is not in the document]
Comment (jesperlauridsen) [reply src/components/SearchResults.vue:17]: Changed.

File: src/components/SearchResults.vue [hunk 8]
```diff
@@ -13,14 +13,19 @@
...
-               <span v-if="resultHits === 1"> pdf</span>
-           </div>
-           
-
-              <component :is="searchResultComponentName"  v-for="(item, index) in this.results" :result="item" :indexNumber="index" :queryString="queryDisplay" :key="index"  ></component>
+             <span class="numbersFound">{{ resultHits || "0" }}</span> matches was found.
+             <span v-if="loading === false && this.results.length > 0">Showing hit 
+              <span class="numbersFound"> {{ this.solrOptions.currentOffset + 1 }} - {{ this.solrOptions.currentOffset + this.solrOptions.shownResultsNumber > resultHits ? resultHits : this.solrOptions.shownResultsNumber + this.solrOptions.currentOffset }}</span>.</span>
+          </div>
+          <div v-if="loading === false && this.results.length > 0" class="pagingButtonContainer">
```
[the remark this answers is not in the document]
Comment (jesperlauridsen) [reply src/components/SearchResults.vue:20]: Changed.

File: src/components/SearchResults.vue [hunk 9]
```diff
@@ -13,14 +13,19 @@
...
+            <button class="pagingButton back" :disabled="this.solrOptions.currentOffset <= 0" v-on:click="previousResults">previous 10</button>
+            <button class="pagingButton forward" :disabled="this.solrOptions.currentOffset + this.solrOptions.shownResultsNumber >= this.resultHits" v-on:click="nextResults">next 10</button>
+          </div>
+          <component :is="searchResultComponentName"  v-for="(item, index) in this.results" :result="item" :indexNumber="index" :queryString="queryDisplay" :key="index"  ></component>
+          <div v-if="loading === false && this.results.length > 0" class="pagingButtonContainer pageEnd">
```
[the remark this answers is not in the document]
Comment (jesperlauridsen) [reply src/components/SearchResults.vue:25]: Changed.

File: src/components/SearchResults.vue (hunk 2 above)
Comment (jesperlauridsen) [reply src/components/SearchResults.vue:27]: Allrighty, i'll leave it for now.

File: src/components/searchresult/common/PDFSearchResult.vue (hunk 1 above)
Comment (jesperlauridsen) [reply src/components/searchresult/common/PDFSearchResult.vue:71]: Yes! I'll do that!

File: src/services/SearchService.js [hunk 10]
```diff
@@ -7,7 +7,7 @@ export const searchService = {
   structureSearchResult,
 }
 
-function fireSearch(query, instance) {
+function fireSearch(query, instance, options, sort) {
     if (query != undefined && query.includes("&d=")) {
```
[the remark this answers is not in the document]
Comment (jesperlauridsen) [reply src/services/SearchService.js:11]: This whole section is a reminiscence of old functionality, and will be removed. How i've missed this is beyond me.

File: src/services/SearchService.js [hunk 11]
```diff
@@ -16,7 +16,7 @@ function fireSearch(query, instance) {
       //searchStore.queryDisplay = query;
     }
     query === undefined ? query = '' : query = "&q=" + query;
```
[the remark this answers is not in the document]
Comment (jesperlauridsen) [reply src/services/SearchService.js:18]: I am :-)

File: src/store/SearchStore.js [hunk 12]
```diff
@@ -22,15 +27,22 @@ const actions = {
...
-
+  updateShownResultsNumber( {commit}, param ) {
+    commit('updateShownResultsNumberSuccess', param)
+  },
+  updateSearchSort( {commit}, param ) {
+    commit('updateSearchSortSuccess', param)
+  },
+  updateCurrentOffset ( {commit}, param ) {
+    commit('updateCurrentOffsetSuccess', param)
+  },
```
[the remark this answers is not in the document]
Comment (jesperlauridsen) [reply src/store/SearchStore.js:41]: Good knowledge to have - We'll keep this for now - I think it's the right direction for us, to keep it consistent throughout the application.

File: src/components/AppliedFilters.vue (hunk 4 above)
[the remark this answers is not in the document]
Comment (jesperlauridsen) [reply src/components/AppliedFilters.vue]: @jorntx Your thoughts :-)

[the remark this answers is not in the document]
Comment (jorntx) [reply src/components/AppliedFilters.vue]: Yes - as long as the return value is ensured to be Boolean - then you can ditch the `=== true`

[the remark this answers is not in the document]
Comment (jesperlauridsen) [reply src/components/AppliedFilters.vue]: Roger - fixed in this instance. Will look out for more of these.

**Pull request merged**